internal/kubernetes/kubernetes.go

```diff
@@ -61,34 +61,36 @@ func prepareLabelSelector(selector string) meta.ListOptions {
 	return listOptions
 }
 
-func (client *KubeClient) ListNamespaces(labelSelector string) []corev1.Namespace {
+func (client *KubeClient) ListNamespaces(labelSelector string) ([]corev1.Namespace, error) {
 	list, err := client.Client.CoreV1().Namespaces().List(context.Background(), prepareLabelSelector(labelSelector))
 
 	if err != nil {
-		logrus.WithError(err).Error("ListNamespaces errored!")
-		return []corev1.Namespace{}
+		return []corev1.Namespace{}, fmt.Errorf("failed to list namespaces")
```

Comment:
Could we please add here the initial error to the message?

Reply:
Rebased on main and added the initial error message.